### ISSUE TITLE
Generic.WhiteSpace.ScopeIndent revertion in 2.6.2

### ISSUE THREAD

**Issue by alasdairdc** [squizlabs/PHP_CodeSniffer#2092]
Have been checking our installed versions and noticed we were still using `2.6.0` (hence looking to update).  Having updated came across what i think may have been a side effect of #1026 as starting with `2.6.2` the following code no longer is flagged as incorrect.

cs-test.php
```php
 <?php

if ($a > $b) {
        $c = $d->my_func(
            $arg1,
            $arg2
        );  

    if ($c > $d) {
        echo 'test';
    }   
}
```

Running with 2.6.1

```
$ ./phpcs-2.6.1.phar --sniffs=Generic.WhiteSpace.ScopeIndent --standard=PSR2 --runtime-set scope_indent_debug true cs-test.php 
Start with token 0 on line 1 with indent 0
Open scope (T_IF) on line 3
	=> indent set to 4 by token 12 (T_OPEN_CURLY_BRACKET)
Closing parenthesis found on line 7
	* first token on line 4 is 15 (T_VARIABLE) *
	=> checking indent of 8; main indent set to 8 by token 15 (T_VARIABLE)
[Line 9] Line indented incorrectly; expected 8 spaces, found 4
	=> Add adjustment of 4 for token 37 (T_IF) on line 9
Open scope (T_IF) on line 9
	=> indent set to 12 by token 47 (T_OPEN_CURLY_BRACKET)
Indent adjusted to 12 for T_ECHO on line 10
	=> Add adjustment of 4 for token 50 (T_ECHO) on line 10
[Line 10] Line indented incorrectly; expected at least 12 spaces, found 8
Close scope (T_IF) on line 11
	=> indent set to 8 by token 56 (T_CLOSE_CURLY_BRACKET)
[Line 11] Line indented incorrectly; expected 8 spaces, found 4
	=> Add adjustment of 4 for token 56 (T_CLOSE_CURLY_BRACKET) on line 11
Close scope (T_IF) on line 12
	=> indent set to 0 by token 58 (T_CLOSE_CURLY_BRACKET)

FILE: .cs-test.php
----------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
----------------------------------------------------------------------
  9 | ERROR | [x] Line indented incorrectly; expected 8 spaces, found
    |       |     4
 10 | ERROR | [x] Line indented incorrectly; expected at least 12
    |       |     spaces, found 8
 11 | ERROR | [x] Line indented incorrectly; expected 8 spaces, found
    |       |     4
----------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 5ms; Memory: 4Mb
```

Running with 2.6.2

```
$ ./phpcs-2.6.2.phar --sniffs=Generic.WhiteSpace.ScopeIndent --standard=PSR2 --runtime-set scope_indent_debug true cs-test.php 
Start with token 0 on line 1 with indent 0
Open scope (T_IF) on line 3
	=> indent set to 4 by token 12 (T_OPEN_CURLY_BRACKET)
Closing parenthesis found on line 7
	* token is inside condition 2 (T_IF) on line 3 *
	* using condition *
	=> checking indent of 4; main indent set to 4 by token 2 (T_IF)
Open scope (T_IF) on line 9
	=> indent set to 8 by token 47 (T_OPEN_CURLY_BRACKET)
Close scope (T_IF) on line 11
	=> indent set to 4 by token 56 (T_CLOSE_CURLY_BRACKET)
Close scope (T_IF) on line 12
	=> indent set to 0 by token 58 (T_CLOSE_CURLY_BRACKET)
```

Other Details:
- OS: Debian 9.4
- PHP version: PHP 7.0.30-0+deb9u1 (cli) (built: Jun 14 2018 13:50:25) - current version from `debian-security`.

**Comment by gsherwood** [comment]
The ScopeIndent sniff has received numerous bug fixes over the years to make it more flexible, less strict by default, and to allow it to support a larger number of coding standards.

I think you're just seeing a fix because the code looks fine to me. The function call is indented at least 4 spaces from the IF (it's indented 8) and the second IF statement is correctly in position (ScopeIndent is more strict about control structures).

The errors you have listed for 2.6.1 are clearly wrong as the IF statement is indented exactly how you'd want. The sniff was previously being tripped up by the double-indented function call.

**Comment by alasdairdc** [comment]
@gsherwood So I follow what you are saying correctly, at `2.6.2` the matching rule changed from an `exact` match to an `atleast` match for anything other than control structure statements (hence the function call not being flagged for too many indents) and as of `2.6.2` is now the intended behaviour.

**Comment by gsherwood** [comment]
> So I follow what you are saying correctly, at 2.6.2 the matching rule changed from an exact match to an atleast match for anything other than control structure statements

No, that rule has always been enforced. That's why there is an `exact` sniff property, although I don't ever recommend people use it because it is very very strict.

The change was that the way indents were set was being done incorrectly because the part that looked at closing parenthesis of function calls *assumed* indents were exact. That's a bad assumption to make if the `exact` flag is not set, so it doesn't do that any more. It uses the rules from the control structure it is in to set the indent level.

Many other parts of the sniff have changed to do this as well. If you are using PHPCS v2, you're missing out on quite a lot of improvements to the ScopeIndent sniff.
